__tests__/pet.test.js

```diff
@@ -0,0 +1,218 @@
+      pet = new Pet('Jessie');
+    });
+    it('returns I need to walk if pets fitness is 3 or less', () => {
+      pet.fitness = 3;
+      pet.checkUp();
+      expect(pet.checkUp()).toEqual('I need a walk')
+    });
+    it('returns I need to walk if pets fitness is 1', () => {
+      pet.fitness = 1;
+      pet.checkUp();
+      expect(pet.checkUp()).toEqual('I need a walk')
+    });
+    it('returns I am hungry if pets hunger is 5 or more', () => {
+      pet.hunger = 5;
+      pet.checkUp();
+      expect(pet.checkUp()).toEqual('I am hungry');
+    });
+    it('returns I am hungry if pets hunger is 9', () => {
+      pet.hunger = 9;
+      pet.checkUp();
+      expect(pet.checkUp()).toEqual('I am hungry');
+    });
+    it('returns I need a walk AND I am hungry if fitness is 3 or less and its hunger is 5 or more', () => {
+      pet.fitness = 3;
+      pet.hunger = 5;
+      expect(pet.checkUp()).toEqual('I need a walk AND I am hungry');
+    });
+    it('returns I need a walk AND I am hungry if fitness is 1 and its hunger is 8', () => {
+      pet.fitness = 1;
+      pet.hunger = 8;
+      expect(pet.checkUp()).toEqual('I need a walk AND I am hungry');
+    });
+    it('returns I feel great if pets fitness is 4 or above and its hunger is 4 or less', () => {
+      pet.fitness = 4;
+      pet.hunger = 4;
+      expect(pet.checkUp()).toEqual('I feel great');
+    });
+    it('returns I feel great if pets fitness is 7 and its hunger is 2', () => {
+      pet.fitness = 7;
+      pet.hunger = 2;
+      expect(pet.checkUp()).toEqual('I feel great');
+    });
+    it('throws an error if the pet is not alive', () => {
+      pet.age = 30;
+      expect(() => pet.checkUp()).toThrow('Your pet is no longer alive :(');
+    });
+  });
+
+  describe('isAlive', () => {
+    let pet;
+    beforeEach(() => {
+      pet = new Pet('Jessie');
+    });
+    it('returns false if fitness is 0 or less', () => {
+      pet.fitness = 0;
+      expect(pet.isAlive).toBe(false);
+    });
+    it('returns false if hunger is 10 or more', () => {
+      pet.hunger = 10;
+      expect(pet.isAlive).toBe(false);
+    });
+    it('returns false if pet age is 30 or more', () => {
+      pet.age = 30;
+      expect(pet.isAlive).toBe(false);
+    });
+    it('returns true if pet fitness is 1 or above, hunger is 9 or less, age is under 30', () => {
+      pet.fitness = 1;
+      pet.hunger = 9;
+      pet.age = 29;
+      expect(pet.isAlive).toBe(true);
+    });
+  });
+
+  describe('adoptChild', () => {
+    let parent;
+    let child;
+    beforeEach(() => {
+      parent = new Pet('Jessie');
+      child = new Pet('Nima');
+    });
+    it('adds child to parent property', () => {
+      parent.adoptChild(child);
+      expect(parent.children).toBeInstanceOf(Array);
+    })
+    it('checks that child is being fed', () => {
+      parent.adoptChild(child);
+      child.hunger = 6;
+      parent.children[0].feed();
+      expect(child.hunger).toEqual(3);
```

Comment:
At first glance this is a bit misleading, as you are manipulating the `child` variable as well as the first element in the `parent.children` array. As it happens, this will still work but it could be made neater by always working on the same variable ie 
```
parent.children[0].hunger = 6;
parent.children[0].feed();
expect(parent.children[0].hunger).toEqual(3);
```